examples/cluster-go/main.go

```diff
@@ -39,10 +45,79 @@ func main() {
 			return err
 		}
 
-		// Export the kubeconfig for clusters
+		//////////////////////////////////////////////////
+		///     Create an ipv6 enabled EKS cluster     ///
+		//////////////////////////////////////////////////
+
+		// 1. Create a VPC with IPv6 CIDR block.
+		vpc, err := ec2.NewVpc(ctx, "ipv6-vpc", &ec2.VpcArgs{
+			AssignGeneratedIpv6CidrBlock: pulumi.Bool(true),
+			EnableDnsSupport:             pulumi.Bool(true),
+			EnableDnsHostnames:           pulumi.Bool(true),
+			CidrBlock:                    pulumi.String("10.100.0.0/16"),
+		})
+		if err != nil {
+			return err
+		}
+
+		// 2. Create two subnets with IPv6 CIDR block in two different availability zones.
+		//    EKS requires at least two subnets in different availability zones.
+		var subnetIDs []pulumi.StringInput
+		for idx, az := range []string{"us-west-2a", "us-west-2b"} {
+			subnet, err := ec2.NewSubnet(ctx, fmt.Sprintf("ipv6-subnet-%d", idx), &ec2.SubnetArgs{
+				VpcId:                       vpc.ID().ToStringOutput(),
+				AssignIpv6AddressOnCreation: pulumi.Bool(true),
+				// EnableDns64:                             pulumi.Bool(true),
+				// EnableResourceNameDnsAaaaRecordOnLaunch: pulumi.Bool(true),
+				AvailabilityZone: pulumi.String(az),
+				CidrBlock:        pulumi.String(fmt.Sprintf("10.100.%d.0/24", len(subnetIDs))),
+				// Ipv6Native:    pulumi.Bool(true), // Create a ipv6 only subnet
+				Ipv6CidrBlock: calculateIPV6CidrBlock(vpc.Ipv6CidrBlock, idx),
+			})
+			if err != nil {
+				return err
+			}
+
+			subnetIDs = append(subnetIDs, subnet.ID().ToStringOutput())
+		}
+
+		// 3. Create an EKS cluster with IPv6 networking enabled.
+		cluster4, err := eks.NewCluster(ctx, "example-cluster-4", &eks.ClusterArgs{
+			IpFamily:         pulumi.StringPtr("ipv6"),
+			VpcId:            vpc.ID().ToStringOutput(),
+			SubnetIds:        pulumi.StringArray(subnetIDs),
+			UseDefaultVpcCni: func() *bool { t := true; return &t }(),
```

Comment:
This line seems unusual, could it be `pulumi.BoolPtr(true)`?

Reply:
`UseDefaultVpcCni` requires a `*bool` type, and that's because we define it as a plain type in schema (ref: https://github.com/pulumi/pulumi-eks/blob/449635cb9dfdfbdc6a03b2d0c3b8b12f96420f13/provider/cmd/pulumi-gen-eks/main.go#L319) when the MLCs were created for this provider. This is an artifact of the nodejs types driving the MLC types, and ideally should really not be a plain type here.